#include "face_frame.h"
#include "ui_face_frame.h"
#include <face_qlabel.h>
#include <QMovie>
#include <QDebug>

Face_Frame::Face_Frame(QWidget *parent) :
    QWidget(parent),
    ui(new Ui::Face_Frame)
{
    ui->setupUi(this);
    setWindowTitle("QQ_Face");

    QMovie *pm = new QMovie(":/resorces/qqface/1.gif");
    ui->label_1->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/2.gif");
    ui->label_2->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/3.gif");
    ui->label_3->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/4.gif");
    ui->label_4->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/5.gif");
    ui->label_5->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/6.gif");
    ui->label_6->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/7.gif");
    ui->label_7->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/8.gif");
    ui->label_8->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/9.gif");
    ui->label_9->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/10.gif");
    ui->label_10->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/11.gif");
    ui->label_11->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/12.gif");
    ui->label_12->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/13.gif");
    ui->label_13->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/14.gif");
    ui->label_14->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/15.gif");
    ui->label_15->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/16.gif");
    ui->label_16->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/17.gif");
    ui->label_17->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/18.gif");
    ui->label_18->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/19.gif");
    ui->label_19->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/20.gif");
    ui->label_20->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/21.gif");
    ui->label_21->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/22.gif");
    ui->label_22->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/23.gif");
    ui->label_23->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/24.gif");
    ui->label_24->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/25.gif");
    ui->label_25->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/26.gif");
    ui->label_26->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/27.gif");
    ui->label_27->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/28.gif");
    ui->label_28->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/29.gif");
    ui->label_29->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/30.gif");
    ui->label_30->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/31.gif");
    ui->label_31->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/32.gif");
    ui->label_32->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/33.gif");
    ui->label_33->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/34.gif");
    ui->label_34->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/35.gif");
    ui->label_35->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/36.gif");
    ui->label_36->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/37.gif");
    ui->label_37->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/38.gif");
    ui->label_38->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/39.gif");
    ui->label_39->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/40.gif");
    ui->label_40->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/41.gif");
    ui->label_41->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/42.gif");
    ui->label_42->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/43.gif");
    ui->label_43->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/44.gif");
    ui->label_44->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/45.gif");
    ui->label_45->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/46.gif");
    ui->label_46->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/47.gif");
    ui->label_47->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/48.gif");
    ui->label_48->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/49.gif");
    ui->label_49->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/50.gif");
    ui->label_50->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/51.gif");
    ui->label_51->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/52.gif");
    ui->label_52->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/53.gif");
    ui->label_53->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/54.gif");
    ui->label_54->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/55.gif");
    ui->label_55->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/56.gif");
    ui->label_56->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/57.gif");
    ui->label_57->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/58.gif");
    ui->label_58->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/59.gif");
    ui->label_59->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/60.gif");
    ui->label_60->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/61.gif");
    ui->label_61->setMovie(pm);

    pm->start();
    pm=new QMovie(":/resorces/qqface/62.gif");
    ui->label_62->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/63.gif");
    ui->label_63->setMovie(pm);
    pm->start();

    pm=new QMovie(":/resorces/qqface/64.gif");
    ui->label_64->setMovie(pm);
    pm->start();

    connect(ui->label_1,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));//将单击的信号链接到相应的槽函数中；
    connect(ui->label_2,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_3,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_4,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_5,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_6,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_7,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_8,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_9,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_10,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_11,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_12,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_13,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_14,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_15,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_16,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_17,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_18,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_19,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_20,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_21,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_22,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_23,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_24,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_25,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_26,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_27,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_28,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_29,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_30,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_31,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_32,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_33,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_34,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_35,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_36,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_37,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_38,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_39,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_40,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_41,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_42,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_43,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_44,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_45,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_46,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_47,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_48,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_49,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_50,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_51,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_52,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_53,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_54,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_55,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_56,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));

    connect(ui->label_57,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_58,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_59,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_60,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_61,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_62,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_63,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
    connect(ui->label_64,SIGNAL(clicked()),this,SLOT(face_qlable_clicked()));
}


//获取被点击QQ表情的QLabel值，用于显示
void Face_Frame::face_qlable_clicked()
{
    Face_QLabel *sendname=qobject_cast<Face_QLabel*>(sender());            //获取被单击的我的myqlabel的对象的名称；
    QString beclicked_name=sendname->objectName();
    qDebug()<<beclicked_name;
    emit display_face_in_dialogbox(beclicked_name);                               //   发送被单击后的对戏那个的控件的名称；

    this->close();
}

Face_Frame::~Face_Frame()
{
    delete ui;
}
